src/main/resources/assets/carpet/scripts/distance_rc_1.sc

```diff
@@ -0,0 +1,82 @@
+import('math','_euclidean','_manhattan','_round');
+
+__config()->{
+    'commands'->{
+        'from <start_pos>'->'set_start',
+        'from <start_pos> to <end_pos>'->'_calculate',
+        'to <end_pos>'->_(end_pos)->(_calculate(null,end_pos)),
+        'toggle_line'->'_toggle_line'
+    },
+    'scope'->'player'
+};
+
+global_position=null;
+global_end_position=null;
+
+global_show_line=false;
+
+set_start(pos)->(
+    global_position= pos;
+    print(player(),format('gi Initial point set to: ', 'g '+pos,'?/tp '+(str(pos)-','-'['-']')))
+);
+
+_toggle_line()->(
+    global_show_line= !global_show_line;
+    if(global_show_line,
+        print('Toggled lines on'),
+        print('Toggled lines off')
+    );
+    null
+);
+
+_calculate(start_pos, end_pos)->(
+    if(start_pos!=null,
+        set_start(start_pos);
+        calculate(end_pos),
+        if(global_position,
+            calculate(end_pos),
+            print(player(),format('r There was no initial point for '+player()));
+            set_start(end_pos)
+        )
+    )
+);
+
+__on_player_right_clicks_block(player, item_tuple, hand, block, face, hitvec) ->
+    pos = pos_offset(block,face);//for clicking against the wall.
+    if(block=='brown_carpet' && system_info('world_carpet_rules'):'carpets'=='true',
+        if(global_position==null||player~'sneaking',//wont complain for first carpet
+            set_start(pos),
+            _calculate(null,pos)
+        )
+    )
+);
+
+calculate(end_pos)->(
+    start_pos = global_position;
+    [dx,dy,dz] = map(start_pos-end_pos,abs(_));
+
+    manhattan = _round(_manhattan([dx, dy, dz],[0,0,0]),100);//rounding to nearest 0.01 for precision, but not ridiculous decimal places
+    spherical = _round(_euclidean([dx, dy, dz],[0,0,0]),100);
+    cylindrical = _round(_euclidean([dx, 0, dz],[0,0,0]),100);//useful for chunk distances etc.
+
+    print(player(),format(
+        'w Distance between ',
+        'c '+start_pos,'?/tp '+(str(start_pos)-','-'['-']'), 'w  and ',
+        'c '+end_pos,'?/tp '+(str(end_pos)-','-'['-']'), 'w :\n',
+        'w  - Spherical: ', 'wb '+spherical+'\n',
+        'w  - Cylindrical: ', 'wb '+cylindrical+'\n',
+        'w  - Manhattan: ', 'wb '+manhattan
+    ));
+
+    global_end_position = end_pos;
+
+    schedule(0,'_draw_line',start_pos,end_pos);
+);
+
+_draw_line(start_pos,end_pos)->(
+    if(global_show_line && global_position == start_pos && global_end_position == end_pos,//so if it changes, it will stop drawing this line.
```

Comment:
Oh, thought having multiple lines was a feature and not a bug, lol. I was quite used to it. nvm.

However, this new change will still create up to infinite lines if the distance is calculated against the same two points over and over (which may seem unlikely, but if someone wants to forcefully lag/crash a server, they just have to spam `/distance to [pos]`).

Would suggest maybe not scheduling a new line on creation but just reuse the same schedule and use the globals instead of locals to draw it. You would have to start the schedule at some point, maybe starting and stopping when running `toggle_line` (starting is scheduling, stopping is not rescheduling, there's still no way to unschedule in Scarpet sadly).
It could still be abused if really willing to, but it's more difficult (to prevent this, maybe make the schedule toggle another bool off and only schedule a new one if that variable got false).

An implementation of the safe one would be like (please don't trust any of the following code more than a bad reference, first of all `global_` being `g_`):
```py
_toggle_line -> (
  if(g_draw_line,
    g_draw_line = false;
  ,!g_draw_line
    g_draw_line = true;
    if(g_stopped,schedule(0,'_draw_line');
    g_stopped = false;
  )
);

draw_line -> (
  if(g_draw_line,
    draw_shape('line',20,'from',global_start_pos,'to',global_end_pos, 'player',player());
    schedule(20,'_draw_line');
  ,
    g_stopped = true
  )
)
```
This one would make sure that the schedule is actually stopped before creating a new one, because else toggle_line could be called multiple times before the schedule actually stops, creating a new one and keeping the existing one since draw would still be on.